### ISSUE TITLE
Old items shown as 'new'

### ISSUE THREAD

**Issue by murkwuite** [ancestorak/brief#134]
Using Brief 2.0b3 in Firefox 35.0.1 on a MacBook Pro, I experience the following problem.

The British newspaper the Guardian has a feed available for its index pages, which a user can find by simply adding '/rss' to the index page URL, as described at http://www.theguardian.com/help/feeds. A problem I've been experiencing with these feeds is that very old items are frequently introduced as if these were new items.

In the image below, Brief can be seen displaying two old items as 'new' for the Guardian's http://www.theguardian.com/global-development/rss feed. Note that at the time these items were actually new (2012 and 2011 respectively), I didn't yet subscribe to these feeds, either using Brief or any other feed reader. So, in a feed that's just been added to Brief for the first time, such 'new' old items show up. Several of these old items will come up on any given day, along with the regular new items.

![schermafbeelding 2015-02-11 om 15 05 07](https://cloud.githubusercontent.com/assets/11019691/6204564/284a1abe-b550-11e4-9cf0-7be6062524e3.png)

Now the above feed URL is the standard one the Guardian encourages people to subscribe to. Interestingly, it turns out there's an alternative URL for at least some of the newspaper's feeds. For the Global Development feed shown above, it's http://feeds.theguardian.com/theguardian/science/rss. The basic technique is to use 'feeds' instead of 'www' and insert '/theguardian' after '.com'. These alternative feeds will work fine, so without coming up with 'new' old items.

I don't have this problem with any other feeds in Brief, only with those from the Guardian. It seems to be a Brief thing, as feeds that display this problem in Brief will work fine in Feedly.

**Comment by tanriol** [comment]
Thank you for this report, this looks strange. I don't see any such items right now, but will try to catch them.

Do you see these only in the Guardian feeds you added long ago or in any recently added "standard" feeds too?

**Comment by murkwuite** [comment]
I've only started following feeds from the Guardian a few weeks ago and have had these problems from the start. If I add other Guardian feeds now, these have this problem as well. It doesn't matter if I unsubscribe from 'misbehaving' feeds and resubscribe, the only thing that works is using the alternative URL.

**Comment by AussieD** [comment]
I have something similar that happens from time to time to a couple of feeds I subscribe to. I just put it down to a quirk associated with the feed itself, rather than any kind of problem specific to Brief. Next time it happens to me I'll check my Subscribed Feeds folder in FF's Bookmarks library to see when the news landed (which, for my thinking, would give insight whether it's just a quirky source feed or a bug with Brief).

**Comment by murkwuite** [comment]
The Guardian technical team has looked into this and haven't been able to find any problems with the the feed at their end. Added to the fact that the misbehaving feeds work fine in Feedly, it does look like this is a Brief thing.

**Comment by tanriol** [comment]
> The Guardian technical team has looked into this and haven't been able to find any problems with the the feed at their end. Added to the fact that the misbehaving feeds work fine in Feedly, it does look like this is a Brief thing.

First, the entries that pop up absolutely do appear in the Guardian feed. Brief has no cloud and no own servers, it cannot get the entry from anywhere except the feed it was given.

I've looked into the feed and I've really seen these old items in it. The exact reason remains unknown to me, but it may be some kind of background updates on the site or so. Anyway, Brief finds an item its user hasn't read yet and correctly marks it as an unread one.

I'm not exactly sure what is the correct behaviour here, whether Brief should ignore the too-old items (that is, items older than... what?) or just leave that to the user as he may expect to be notified of these items.

**Comment by murkwuite** [comment]
This user, for one, certainly does not expect to be notified of years-old items, which tend to rather clog up his feeds (there are a lot of those old items in them!). It's interesting that Feedly doesn't display them, showing only genuinely new items. And it's curious that the misbehaving feeds have an alternatively formed URL that does work well (as explained in my original post).

The title of this thread has just been changed to include 'multi-year-old items'. Please note that not all of the old items are that old, though many are. Some older items appearing only in the misbehaving feeds are just a few weeks old.

**Comment by AussieD** [comment]
The couple of 'misbehaving' feeds that I have which I referred to earlier were, in fact, dumping old news items into my Subscribed Feeds folder (as suspected). Tanroil is quite right to say that the problem is not with Brief, but the source feed, itself.

Feedly is a cloud managed RSS service. The adminstrators of that service/platform may have developed in-built functionality that automatically manages what is happening in those instances. Brief, on the other hand, is a local Add-in that can only see (read) what has already landed in your Subscribed Feeds RSS Live Bookmarks folder.

I think the issues you are experiencing are a by-product of adding /rss to the URL of an index page, as recommended by The Guardian (http://www.theguardian.com/help/feeds). If the source index page (e.g. http://www.theguardian.com/global-development) is updated by a Guardian sub-editor to include any older news items in its index, then those new additions of old news items would presumably appear in your RSS feed (because it represents new information as far are your customised RSS feed is concerned).

Basically, I'm suggesting that adminstrative behind-the-scenes changes to a source news index page that add old content to an index page might be pushed out as new *feed content* to FF at that time (because it has been recently added to the index). If this is correct, then it amounts to an unavoidable by-product of adding /rss to a news index page. It would also mean that, although the Guardian's recommended approach works, its silence on the possible side-effects may not be serving you well.

It's possible the devs of Brief could add functionality for end-users to filter out news items that are older than "x" age (which tanriol has already alluded to), but since I'm not a dev I'll leave that to them.

**Comment by murkwuite** [comment]
I've contacted the Guardian technical team again to let them know the problem really does appear to be with their feeds, notwithstanding their original response. I haven't heard back from them.

All the same, it might be worthwile considering adding an option to Brief that lets the user filter out messages older than X days.
